auth/jwt.go

```diff
@@ -0,0 +1,197 @@
+`
+
+const devPublicKey = `-----BEGIN PUBLIC KEY-----
+MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEteZmZdBkEXr+MruFSHYDs75ArTDx
+ouUDFsC1WEmAkbfh7zUp6ybM9a54lQNkHZ2uW7pQ+9JkqTzvYU+pt7uAMg==
+-----END PUBLIC KEY-----
+`
+
+const refreshTokenDuration = time.Hour * 24 * 7
+const accessTokenDuration = time.Minute * 10
+
+var tokenPrivateKey *ecdsa.PrivateKey
+var tokenPublicKey *ecdsa.PublicKey // nolint:unused
+
+func SetupTokenSecrets(mode string, privateKey string, publicKey string) {
+	if privateKey == "" || publicKey == "" {
+		if mode == gin.ReleaseMode {
+			zerolog.Fatal().Msg("could not find jwt-private.pem and/or jwt-public.pem")
+		} else {
+			privateKey = devPrivateKey
+			publicKey = devPublicKey
+			zerolog.Warn().Msg("using the dev keys for JWTs")
+		}
+	}
+
+	private, err := jwt.ParseECPrivateKeyFromPEM([]byte(privateKey))
+	if err != nil {
+		zerolog.Fatal().Err(err).Msg("could not parse private key")
+	}
+	tokenPrivateKey = private
+
+	public, err := jwt.ParseECPublicKeyFromPEM([]byte(publicKey))
+	if err != nil {
+		zerolog.Fatal().Err(err).Msg("could not parse public key")
+	}
+	tokenPublicKey = public
+}
+
+type UserOrOrgClaim struct {
+	ID   uuid.UUID `json:"id" example:"d45ddfc0-1fc3-4fa7-95e9-edd7d134103a"`
+	Role string    `json:"role" example:"user"`
+}
+
+type TokenClaim struct {
+	Exp          time.Time      `json:"exp"`
+	ID           uuid.UUID      `json:"id"`
+	User         UserOrOrgClaim `json:"user"`
+	Organization UserOrOrgClaim `json:"organization"`
+}
+
+func (claim TokenClaim) Valid() error {
+	if claim.Exp.Before(time.Now()) {
+		return jwt.ErrTokenExpired
+	}
+	return nil
+}
+
+type RefreshTokenClaim struct {
+	Type string `json:"type"`
+	TokenClaim
+}
+
+func (claim RefreshTokenClaim) Valid() error {
+	if claim.Type != "refresh" {
+		return errors.New("not a refresh token")
+	}
+	return claim.TokenClaim.Valid()
+}
+
+type AccessTokenClaim struct {
+	Type string `json:"type"`
+	TokenClaim
+}
+
+func (claim AccessTokenClaim) Valid() error {
+	if claim.Type != "access" {
+		return errors.New("not an access token")
+	}
+	return claim.TokenClaim.Valid()
+}
+
+func IssueRefreshToken(logCtx context.Context, user UserOrOrgClaim, org UserOrOrgClaim) (string, RefreshTokenClaim, error) {
+	id := uuid.New()
+	exp := time.Now().Add(refreshTokenDuration)
+
+	claims := RefreshTokenClaim{
+		Type: "refresh",
+		TokenClaim: TokenClaim{
+			Exp:          exp,
+			ID:           id,
+			User:         user,
+			Organization: org,
+		},
+	}
+
+	token := jwt.NewWithClaims(jwt.SigningMethodES256, claims)
+	signedString, err := token.SignedString(tokenPrivateKey)
+	if err != nil {
+		return "", claims, err
+	}
+
+	zerolog.Ctx(logCtx).Info().
+		Str("user_id", claims.User.ID.String()).
+		Str("refresh_token_id", claims.ID.String()).
+		Str("exp", claims.Exp.String()).
+		Msg("issued new refresh token")
+
+	return signedString, claims, nil
+}
+
+// validateRefreshToken
+func _(tokenString string) (*RefreshTokenClaim, error) {
```

Comment:
Why do you name this functions with an underscore instead of the commented name above?

Reply:
linter complains (dead code)